static-site/src/components/ListResources/Modal.tsx

```diff
@@ -164,7 +178,8 @@ function _draw(ref, data) {
 
   /* Create the x-scale and draw the x-axis */
   const x = d3.scaleTime()
-    .domain([flatData[0].date, new Date()]) // the domain extends to the present day
+    // presence of dates on resource has already been checked so this assertion is safe
```

Comment:
Bracket notation returns `undefined` if accessing out of bounds (IIRC), so the type of `a` is potentially `undefined`.

The problem with comments like this:
```ts
// presence of dates on resource has already been checked so this assertion is safe
```
is that 3 months from now a fresh intern comes and inserts new code between the check and the assertion and then :boom:. Though, to be fair, it did not happen with js previously (due to lack of interns?)

Again, a small wrapper would make it safe and clean.

If you can find a fallback value, then something like this might work:

```ts
flatData[0]?.date ?? new Date()
``` 

This cannot fail and requires no hacks.